custom_components/ramses_cc/__init__.py

```diff
@@ -264,44 +242,11 @@ def __init__(
     @property
     def available(self) -> bool:
         """Return True if the sensor is available."""
-        return getattr(self._device, self._state_attr) is not None
+        return getattr(self.rf_entity, self._state_attr) is not None
 
     @property
     def name(self) -> str:
         """Return the name of the binary_sensor/sensor."""
-        if not hasattr(self._device, "name") or not self._device.name:
-            return f"{self._device.id} {self._state_attr}"
-        return f"{self._device.name} {self._state_attr}"
-
-
-class RamsesZoneBase(RamsesEntity):  # for: climate & water_heater
-    """Base for any RAMSES RF-compatible entity (e.g. Controller, DHW, Zones)."""
-
-    _attr_precision: float = PRECISION_TENTHS
-    _attr_temperature_unit: str = UnitOfTemperature.CELSIUS
-
-    def __init__(self, broker, device) -> None:
-        """Initialize the sensor."""
-        super().__init__(broker, device)
-
-        # dont include platform/domain (climate.ramses_cc)
-        self._attr_unique_id = device.id
-
-    @property
-    def current_temperature(self) -> float | None:
-        """Return the current temperature."""
-        return self._device.temperature
-
-    @property
-    def extra_state_attributes(self) -> dict[str, Any]:
-        """Return the integration-specific state attributes."""
-        return {
-            **super().extra_state_attributes,
-            "schema": self._device.schema,
-            "params": self._device.params,
```

Comment:
The relevant config/params are exposed separately by the relevant entities so this was redundant and somewhat confusing. Technically a breaking change, but it would seem unlikely that anyone is relying on it.

Reply:
Not sure - I'd need to have a think. 

Leave in for now.